sources/dynamodb/streaming.go

```diff
@@ -0,0 +1,47 @@
+package dynamodb
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbiface"
+)
+
+// NewDynamoDBStream initializes a new DynamoDB Stream for a table with NEW_IMAGE
+// StreamViewType. If there was already an enabled stream for a given table
+// then it first disables it and creates a new one. It returns latest Stream
```

Comment:
> Is the user asked or notified that this will be done by the tool? This can have an impact of the other user workflows which are reliant on the DynamoDB stream.

Before bulk migration there will be a message on the command line interface that "Starting DynamoDB Streams initialization". We can change it to be more clear that "Creating new DynamoDB Streams for tables".

> What happens if the stream exists with NEW_AND_OLD_IMAGES and the user has built some functionality on it?

This is a good corner case which we need to address. For processing purposes in our solution only NewImage and Keys will be required. Because of that NEW_IMAGE was used because there was no need for OldImage.

We can use NEW_AND_OLD_IMAGES as it stores all the things like NewImage, OldImage, Keys. This will not impact the customers as even if they are using DynamoDB Streams for some purpose then they will still have all the information available which they were using earlier.

Reply:
> Before bulk migration there will be a message on the command line interface that "Starting DynamoDB Streams initialization".

That is just a log message right? I think that will be easy to miss and does not give much information to the user about the changes that HarbourBridge would make behind the scenes to the table streams.

> This is a good corner case which we need to address. For processing purposes in our solution only NewImage and Keys will be required. Because of that NEW_IMAGE was used because there was no need for OldImage.

Yes, I think this needs a bit more discussion. As per me, we should be doing something like this:
1. Check if the table has a stream enabled or not.
2. If no, go ahead and enable the stream with type `NEW_AND_OLD_IMAGES`.
3. If yes, check the `StreamViewType`. 
4. If its value is one of `[KEYS_ONLY, OLD_IMAGE]`, then `skip` the table from the migration, print an appropriate log message (asking the user to switch the `StreamViewType` and re-run) and capture the same in the report generated in the end. (Note: We can also prompt the user to do this change on their behalf but I want to hold off on that thought for later for reasons unrelated to this change)
5. If its value is one of `[NEW_AND_OLD_IMAGES, NEW_IMAGE]` proceed to process the stream.

I think I am overall uncomfortable with HarbourBridge making a change to the customer resource which can lead to side-effects for the user outside of the migration.

Reply:
> If its value is one of [KEYS_ONLY, OLD_IMAGE], then skip the table from the migration, print an appropriate log message (asking the user to switch the StreamViewType and re-run) and capture the same in the report generated in the end. (Note: We can also prompt the user to do this change on their behalf but I want to hold off on that thought for later for reasons unrelated to this change)

NEW_AND_OLD_IMAGES is like a superset of all the stream view types. Current way DynamoDB Streams are used by customers is by AWS Lambda Triggers and KCL. So even if we create a new stream with NEW_AND_OLD_IMAGES that will still contain all the information for lambda triggers or KCL to do what they were doing earlier. It's just that some extra information will be added to the record. Am I missing something here or does it make sense?

Reply:
There can be unintended consequences of such a change, since the user used `KEYS_ONLY` or `OLD_IMAGE` for a specific reason earlier.
One scenario that comes to mind - Since NEW_AND_OLD_IMAGES would contain both the old and new image, the size of streaming record would increase significantly, this might affect the billing for the user for their usage of a Lambda trigger via the DynamoDB stream 🤔

Reply:
Yes billing will increase but isn't that implied that we are starting DynamoDB Streams for migration. If some customer wasn't using Streams then they must be aware that this migration uses DynamoDB Streams and it will increase costs. So if some customer is using it then they want to face that cost for migrating data.

Reply:
Yes, you are right. Your example is a more suitable one of the problem I have in mind than the one I have given!